modules/common/src/test/scala/com.snowplowanalytics.snowplow.rdbloader.common/ShredderConfigSpec.scala

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2012-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.common
+
+import java.net.URI
+import java.nio.file.{Paths, Files}
+
+import scala.concurrent.duration._
+
+import com.snowplowanalytics.iglu.core.SchemaCriterion
+
+import com.snowplowanalytics.snowplow.rdbloader.common.config.{Region, ShredderConfig}
+
+import org.specs2.mutable.Specification
+
+class ShredderConfigSpec extends Specification {
+  import ShredderConfigSpec._
+
+  "batch fromString" should {
+    "be able to parse extended batch shredder config" in {
+      val result = getConfig("/shredder.batch.config.reference.hocon", ShredderConfig.Batch.fromString)
+      val expected = ShredderConfig.Batch(
+        exampleBatchInput,
+        exampleOutput,
+        exampleSQSConfig,
+        exampleFormats,
+        exampleMonitoring
+      )
+      result must beRight(expected)
+    }
+
+    "be able to parse minimal batch shredder config" in {
+      val result = getConfig("/shredder.batch.config.minimal.hocon", testParseBatchConfig)
+      val expected = ShredderConfig.Batch(
+        exampleBatchInput,
+        exampleDefaultOutput,
+        exampleSNSConfig,
+        exampleDefaultFormats,
+        exampleDefaultMonitoring
+      )
+      result must beRight(expected)
+    }
+
+    "give error when unknown region given" in {
+      val result = getConfig("/test.config1.hocon", ShredderConfig.Batch.fromString)
+      result.fold(
+        err => err.contains("unknown-region-1"),
+        _ => true
```

Comment:
I realized I made mistake in there. It should be like this:
```suggestion
     _ => false
```
Right means that config parsed successfully which we don't expect it here because we are expecting some error related with region. I will make necessary changes.